.ci/test.sh

```diff
@@ -100,7 +100,7 @@ fi
 if [[ $TASK == "lint" ]]; then
     pwsh -command "Install-Module -Name PSScriptAnalyzer -Scope CurrentUser -SkipPublisherCheck"
     echo "Linting PowerShell code"
-    pwsh -file "./.ci/lint-powershell.ps1" || exit 0
+    pwsh -file "./.ci/lint-powershell.ps1" || :
```

Comment:
Just noticed that after [moving powershell linter upper all other ones](https://github.com/microsoft/LightGBM/pull/6704#discussion_r1821440350), `|| exit 0` causes terminating the whole script earlier than running all linters.

Ref. for `:`
- https://stackoverflow.com/a/33427572.

Reply:
ah right! Sorry 😅 

`|| true` would work too. I'm indifferent, since this is temporary anyway

Reply:
I checked the `lint` job logs ([build link](https://github.com/microsoft/LightGBM/actions/runs/11620094860/job/32361143252?pr=6709)) and can confirm all the other linting is running, so looks like this works. Thanks very much for the Stack Overflow link, I hadn't seen `|| :` before.

Reply:
Oh and I hadn't yet clicked that S/O link before suggesting `|| true`. I see it says there:

> *The above suggestions to use 'true' will also work, but are inefficient as 'true' is an external program.*

That's good to know! I have been using `|| true` as a habit for a long tim.

Reply:
> I have been using `|| true` as a habit for a long tim.

Me too. And hah, seems it's OK - just noticed the following comment under the S/O answer:
> True is only an external program if you run `/bin/true`, as `true` on its own is a shell built-in of bash, just like echo and test, which also exists as external programs but are also built-ins in bash and if you don't give a full path, the built-ins will be used. 

https://stackoverflow.com/questions/17830326/ignoring-specific-errors-in-a-shell-script/33427572#comment69729658_33427572

Reply:
oh wow!